### PR TITLE
Bisection

### PR DESCRIPTION
**Pull request by eggrobin**
1-ULP generic bisection.

### CODE REVIEW

File: geometry/sign.hpp [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Sign {
   bool Negative() const;
   bool Positive() const;
 
+  bool operator==(Sign const& other) const;
```
Comment (pleroy) [inline geometry/sign.hpp]: Should we also have `!=`?

File: numerics/root_finders.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace principia {
+namespace numerics {
+
+// Approximates a root of |f| between |lower_bound| and |upper_bound| by
```
Comment (pleroy) [inline numerics/root_finders.hpp]: Mention that lower_ and upper_bound must have values of different signs and the function must be monotonic.

File: numerics/root_finders_body.hpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+template<typename Argument, typename Function>
+Argument Bisect(Function f,
+                Argument const& lower_bound,
+                Argument const& upper_bound) {
+  using Value = decltype(f(lower_bound));
+  Value const zero{};
+  Value f_upper = f(upper_bound);
+  Value f_lower = f(lower_bound);
+  CHECK(f_lower > zero &&  zero > f_upper || f_lower < zero &&  zero < f_upper);
```
Comment (pleroy) [inline numerics/root_finders_body.hpp]: Move inside the loop?

File: geometry/sign.hpp (hunk 1 above)
Comment (eggrobin) [inline geometry/sign.hpp]: Done.

File: numerics/root_finders.hpp (hunk 2 above)
Comment (eggrobin) [inline numerics/root_finders.hpp]: That's not it, but documented the contract.

File: numerics/root_finders_body.hpp (hunk 3 above)
Comment (eggrobin) [inline numerics/root_finders_body.hpp]: wat.

**Pull request merged**